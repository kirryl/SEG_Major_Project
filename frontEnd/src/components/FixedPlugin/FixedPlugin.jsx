import React, { Component } from "react";

//import Button from "components/CustomButton/CustomButton.jsx";

class FixedPlugin extends Component {
  constructor(props) {
    super(props);
    this.state = {
      classes: "dropdown show"
    };
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick() {
    if (this.state.classes === "dropdown") {
      this.setState({ classes: "dropdown show" });
    } else {
      this.setState({ classes: "dropdown" });
    }
  }
  render() {
    return (
      <div className="fixed-plugin">
        <div className={this.state.classes}>
          <div onClick={this.handleClick}>
            <i className="fa fa-cog fa-2x" />
          </div>
          <ul className="dropdown-menu show">

          </ul>
        </div>
      </div>
    );
  }
}

export default FixedPlugin;
