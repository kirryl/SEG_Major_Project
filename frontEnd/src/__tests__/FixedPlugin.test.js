import renderer from 'react-test-renderer';
import FixedPlugin from '../components/FixedPlugin/FixedPlugin';
import React from 'react';
import { shallow } from 'enzyme';
import enzymeConfig from '../../enzymeConfig'

describe('<FixedPlugin />', function(){

    it('renders correctly', function(){
        const component = renderer.create(<FixedPlugin />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})
