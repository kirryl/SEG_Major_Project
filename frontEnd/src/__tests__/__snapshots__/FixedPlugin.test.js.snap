// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FixedPlugin /> renders correctly 1`] = `
<div
  className="fixed-plugin"
>
  <div
    className="dropdown show"
  >
    <div
      onClick={[Function]}
    >
      <i
        className="fa fa-cog fa-2x"
      />
    </div>
    <ul
      className="dropdown-menu show"
    />
  </div>
</div>
`;

exports[`<FixedPlugin /> should render correctly 1`] = `
<div
  className="fixed-plugin"
>
  <div
    className="dropdown show"
  >
    <div
      onClick={[Function]}
    >
      <i
        className="fa fa-cog fa-2x"
      />
    </div>
    <ul
      className="dropdown-menu show"
    />
  </div>
</div>
`;
